### ISSUE TITLE
compile failures but package still installs

### ISSUE THREAD

**Issue by cbm755** [catch22/octave-doctest#134]
See #133 for the exact error, but the point here is that the package seems to install, even following a compile failure:
````
/usr/include/stdint.h:31: confused by earlier errors, bailing out
g++: error: doctest_evalc.o: No such file or directory
For information about changes from previous versions of the doctest package, run 'news doctest'.
````
Note the "For information about..." message.

(Of course, the package is not functional).

**Comment by cbm755** [comment]
Hmmm, so I tried to reproduce this by putting a deliberate error in `doctest_evalc.cc` but it fails correctly for me:
````
>> pkg install doctest-0.4.1.tar.gz
doctest_evalc.cc:21:1: error: ‘wtf’ does not name a type
 wtf
 ^
In file included from /usr/include/octave-4.0.0/octave/../octave/defun-dld.h:30:0,
                 from /usr/include/octave-4.0.0/octave/../octave/oct.h:36,
                 from doctest_evalc.cc:18:
doctest_evalc.cc: In function ‘octave_function* Gdoctest_evalc(const octave_shlib&, bool)’:
/usr/include/octave-4.0.0/octave/../octave/defun-int.h:98:36: error: ‘Fdoctest_evalc’ was not declared in this scope
   DEFINE_FUNX_INSTALLER_FUN(#name, F ## name, G ## name, doc)
                                    ^
/usr/include/octave-4.0.0/octave/../octave/defun-int.h:112:65: note: in definition of macro ‘DEFINE_FUNX_INSTALLER_FUN’
         octave_dld_function *fcn = octave_dld_function::create (fname, shl, name, doc); 
                                                                 ^
/usr/include/octave-4.0.0/octave/../octave/defun-dld.h:61:3: note: in expansion of macro ‘DEFINE_FUN_INSTALLER_FUN’
   DEFINE_FUN_INSTALLER_FUN (name, doc) \
   ^
doctest_evalc.cc:23:1: note: in expansion of macro ‘DEFUN_DLD’
 DEFUN_DLD (doctest_evalc, args, nargout,
 ^
make: *** [doctest_evalc.oct] Error 1
make: Entering directory '/tmp/oct-0gxRsh/doctest-0.4.1/src'
/usr/bin/mkoctfile-4.0.0 doctest_evalc.cc
Makefile:4: recipe for target 'doctest_evalc.oct' failed
make: Leaving directory '/tmp/oct-0gxRsh/doctest-0.4.1/src'

pkg: error running `make' for the doctest package.
error: called from 'configure_make' in file /usr/share/octave/4.0.0/m/pkg/private/configure_make.m near line 96, column 9
````
(and indeed the package is not installed).

So I cannot reproduce, at least not in this way...

**Comment by cbm755** [comment]
Closing, cannot reproduce.
